src/number_theory/primorial.lean

```diff
@@ -0,0 +1,267 @@
+import data.list
+import data.nat.prime
+import data.nat.choose
+import algebra.euclidean_domain
+import algebra.big_operators
+import tactic
+
+open_locale big_operators
+
+-- This is going into stdlib
+lemma sum_range_choose_halfway (m : nat) : ∑ i in finset.range (m + 1), nat.choose (2 * m + 1) i = 4 ^ m :=
+   by sorry
+
+private lemma add_one_zero (a : nat) (p : a + 1 = 0) : false := by linarith
+
+private lemma mod_2 (n : nat) : (n % 2 = 0) ∨ ((n + 1) % 2 = 0) :=
+begin
+  induction n,
+  {left, refl},
+  cases n_ih,
+  {right, exact n_ih},
+  {left, exact n_ih},
+end
+
+private lemma twice_m_sub_m (m : nat) : 2 * m - m = m :=
+begin
+  rw two_mul m,
+  exact nat.add_sub_cancel m m,
+end
+
+lemma choose_symmetric
+  (m : nat)
+  : nat.choose (2 * m + 1) (m + 1) = nat.choose (2 * m + 1) m
+  :=
+begin
```

Comment:
Might it be nicer to do this using `nat.choose_symm_eq` or the variants near it?

Reply:
Oh, that's infinitely nicer

Reply:
Sorry, I mean `choose_symm_of_eq_add`